app/logic/Mail/EWS/EWSAccount.ts

```diff
@@ -203,7 +203,24 @@ export class EWSAccount extends MailAccount {
     let response = await appGlobal.remoteApp.postHTTP(this.url, this.request2XML(aRequest), "text", this.createRequestOptions());
     response.responseXML = this.parseXML(response.data);
     if (response.status == 200) {
-      return this.checkResponse(response, aRequest);
+      try {
+        return this.checkResponse(response, aRequest);
+      } catch (ex) {
+        if (ex.type != "ErrorServerBusy") {
+          throw ex;
+        }
+        let milliseconds = ex.error.MessageXml?.Value?.Value;
+        if (!milliseconds) {
+          // Server isn't asking us to back off, so give up.
+          throw ex;
+        }
+        if (milliseconds > 20000) {
+          milliseconds = 20000; // Sensible upper limit
+        }
+        console.log(`Server busy, using ${milliseconds}ms backoff time`);
+        await new Promise(resolve => setTimeout(resolve, milliseconds));
```

Comment:
```suggestion
        `sleep(seconds)`;
```

Reply:
You can't sleep in the UI thread.

Reply:
I was referring to our `sleep()` function in util.js, which does exactly what you do here.